src/main/java/com/ss/training/lms/controller/BorrowerController.java

```diff
@@ -92,23 +93,18 @@
 	public ResponseEntity<BookLoan> returnBook(@RequestBody BookLoan loan) {
         if(loan.getBookId() == null || loan.getBranchId() == null || loan.getCardNo() == null)
             return new ResponseEntity<BookLoan>(loan, HttpStatus.BAD_REQUEST);
-        HttpStatus status = HttpStatus.OK;
-        try {
-            borrowerService.returnBook(loan);
-        } catch (ClassNotFoundException | SQLException e) {
-            status = HttpStatus.INTERNAL_SERVER_ERROR;
-        }
-        return new ResponseEntity<BookLoan>(loan, status);
+
+        boolean doesLoanExist = borrowerService.checkIfLoanExists(loan);
+        if(!doesLoanExist)
```

Comment:
It would be good to also check if the book has been returned yet,

Reply:
absolutely, will add that now.